apps/store/src/services/Tracking/internalEvents.ts

```diff
@@ -0,0 +1,39 @@
+import { useCallback } from 'react'
+import type { EventInput } from '@/services/graphql/generated'
+import { useSendEventBatchMutation } from '@/services/graphql/generated'
+import { useShopSession } from '@/services/shopSession/ShopSessionContext'
+import type { TrackingEvent } from '@/services/Tracking/Tracking'
+import { isBrowser } from '@/utils/env'
+
+type EventData = {
+  type: TrackingEvent
+  data: unknown
+}
+
+const eventBuffer: Array<Omit<EventInput, 'sessionId'>> = []
+export const reportInternalEvent = (event: EventData) => {
+  if (!isBrowser()) throw new Error('Must be used client-side')
+  eventBuffer.push({
+    id: crypto.randomUUID(),
+    clientTimestamp: new Date().toISOString(),
+    ...event,
+  })
+}
+
+export const useFlushInternalEvents = () => {
```

Comment:
I'm slightly confused with `flush` in the name :)
```suggestion
export const useSendInternalEvents = () => {
```